gnovm/pkg/gnolang/type_check.go

```diff
@@ -215,6 +215,122 @@ func assertAssignableTo(xt, dt Type, autoNative bool) {
 	}
 }
 
+func checkConstantExpr(store Store, last BlockNode, vx Expr) {
+Main:
+	switch vx := vx.(type) {
+	case *NameExpr:
+		t := evalStaticTypeOf(store, last, vx)
+		if _, ok := t.(*ArrayType); ok {
+			break Main
```

Comment:
currently, the check happens both in `checkConstantExpr` and `declOrDefine`,  can you merge the logic?

I think this logic is not correct, e.g. 
```go
package main

func main() {
	const a = [2]int{1, 2}
	println(a)
}
```
in the current logic, `checkConstantExpr` does not catch the issue, and it's left to the `declOrDefine`..

Reply:
I implemented specific checks in the definedecl function for optimization, but I think they do not enhance readability or significantly improve performance.

Move all const expresion check in one function here: https://github.com/gnolang/gno/pull/2828/commits/1ba4872c2badb651a8212a730245e1d86083181d